service/storage_service.cc

```diff
@@ -2683,20 +2694,19 @@ future<> storage_service::join_cluster(sharded<db::system_distributed_keyspace>&
         auto tmlock = co_await get_token_metadata_lock();
         auto tmptr = co_await get_mutable_token_metadata_ptr();
         for (const auto& [host_id, st] : loaded_endpoints) {
-            if (st.endpoint == get_broadcast_address()) {
+            if (host_id == my_host_id()) {
```

Comment:
Why did you reverse the cases? Previously the host_id conflict case would cause on_internal_error and the IP conflict case would cause a remove_endpoint, now it's reversed.

The previous logic made more sense to me.

I think the IP conflict case can actually happen, although it's very exotic. Suppose we have node A with IP X and node B with IP Y, first node B changes its IP to Z, then node A may change its IP to Y.

Reply:
makes sense.  I'll revert the order to the original one.
(and for the record, I hate the fact system.peers uses the ip address as key, rather than the host_id.  it's backward and we should get rid of that - maybe when we get rid of vnodes.  the sooner the better)

Reply:
unreversed in v11